### PR TITLE
-Fixed a situation where additional emails could be duplicated when crea...

### PR DESCRIPTION
**Pull request by Hiperzone**
...ting a new key

-If CreateKeyEmailFragment view is recreated, the array of additional emails won't be repopulated again if its not null, avoiding new reallocations.
-if CreateKeyEmailFragment view is recreated, the email adapter wont be recreated if its not null, avoiding new reallocations.

### CODE REVIEW

**Comment by dschuermann** [comment]
Nice! Please fix the coding style. Will merge then.

**Comment by Hiperzone** [comment]
Fixed code style. Hopefully everything is ok now :)

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java [hunk 1]
```diff
@@ -144,6 +160,21 @@ public void handleMessage(Message message) {
...
+                    //check for duplicated emails inside the adapter
+                    for (EmailAdapter.ViewModel model : mAdditionalEmailModels) {
+                        if (email.equals(model.email)) {
+                            notifyUser(getString(R.string.create_key_email_already_exists_text));
+                            return;
+                        }
+                    }
+
                     // add new user id
                     mEmailAdapter.add(
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java]: please use already defined variable ``email`` here

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java [hunk 2]
```diff
@@ -122,21 +123,36 @@ public void onClick(View v) {
...
 
         return view;
     }
 
+    /**
+     * Notifies the user with a message
+     *
+     * @param msg
+     */
+    private void notifyUser(String msg) {
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java]: remove this method and use Notify class directly

File: OpenKeychain/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -629,6 +629,7 @@
     <string name="create_key_edit">"Change key configuration"</string>
     <string name="create_key_add_email">"Add email address"</string>
     <string name="create_key_add_email_text">"Additional email addresses are also associated to this key and can be used for secure communication."</string>
+    <string name="create_key_email_already_exists_text">"Email already exists"</string>
```
Comment (dschuermann) [inline OpenKeychain/src/main/res/values/strings.xml]: Maybe "Email has already been added" is better here

**Comment by dschuermann** [comment]
I added some more comments. After fixing those please squash your commits.

**Comment by Hiperzone** [comment]
Will work on it once i finish my proposal :)

**Comment by Hiperzone** [comment]
Hi there, you can close this one, had some troubles with the commit squash and ended up destroying my repository -.- Was the first time i used this feature.

**Comment by dschuermann** [comment]
Merged here: https://github.com/open-keychain/open-keychain/pull/1187

**Pull request closed without merging**